test/jsn_tests.erl

```diff
@@ -499,6 +499,32 @@ path_elements_test_() ->
      ?_assertError(badarg, jsn:path_elements(0)),
      ?_assertError(badarg, jsn:path_elements([10, <<"foo">>]))].
 
+select_test_() ->
```

Comment:
Good start on the tests for this functionality. Please make sure you test the error conditions in your tests, too.